### PR TITLE
Linked list cycle

### PR DESCRIPTION
**Pull request by Hiba-alkurd**
detect linked list cycle solution
I used the hare and tortoise approach
I used two indexes to traverse the linked list one faster than the other

### CODE REVIEW

File: Problems/Problems/LinkedListCycle.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
+﻿using System;
+namespace Problems
+{
+
+    public class LinkedListCycle
+    {
+        void AddCycle(Node head, int pos)
+        {
+            if (pos == -1) return;
```
Comment (ziadadeela) [inline Problems/Problems/LinkedListCycle.cs]: Let's use full names for variables => position (I guess ? 🙄)

File: Linked List Cycle/Main.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        if (head == null) return false;
+        //one node
+        if (head.next == null) return false;
+        ListNode x = head;
+        ListNode y = head.next;
+        while (y!=null)
+        {
+            if (x == y) return true;
+            //one step
+            if (x.next != null) x = x.next;
```
Comment (ziadadeela) [inline Linked List Cycle/Main.cs:25]: Let's have the if body in a separate line, to make it more readable.
if (x.next != null) 
 x = x.next;

Comment (ziadadeela) [reply Linked List Cycle/Main.cs:25]: We can use early return here

**Review by ziadadeela — CHANGES_REQUESTED** [review]

File: Linked List Cycle/Main.cs (hunk 2 above)
Comment (Hiba-alkurd) [reply Linked List Cycle/Main.cs:25]: hmmm yeah by inverting the if statements, good note thanks

**Review by ziadadeela — APPROVED** [review]

**Pull request merged**